app/components/listing/MobileActionBar.jsx

```diff
@@ -2,27 +2,45 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 import { Link } from 'react-router';
+import { getResourceActions } from 'utils/ResourceActions';
 import { images } from '../../assets';
 
+import './MobileActionBar.scss';
+
+const getMobileActions = resource => {
+  const resourceActions = getResourceActions(resource);
+  const mobileActions = [
+    { ...resourceActions.edit, icon: 'edit-blue' },
+  ];
+  if (resource.address) {
+    mobileActions.unshift({ ...resourceActions.directions, icon: 'directions-blue' });
```

Comment:
Also non-blocking comment: I think this `unshift`ing logic would be easier to follow if we stuck to just `push`ing onto the array. Another way we could write this is to initialize an empty array, and then conditionally append in order rather than prepending. E.g.

```jsx
const mobileActions = [];

if (resource.phones && resource.phones.length > 0) {
  mobileActions.push({ ... });
}

if (blah) {
  mobileActions.push(blah);
}

mobileActions.push(thingThatShouldAlwaysBePresent);
```